haskell/private/packages.bzl

```diff
@@ -54,10 +58,19 @@ def expose_packages(build_info, lib_info, use_direct, use_my_pkg_id, custom_pack
             # XXX: use_my_pkg_id is not None only in compile.bzl
             if (use_my_pkg_id == None) or package != use_my_pkg_id:
                 args.extend(["-package-id", package])
+                package_ids.append(package)
 
     # Only include package DBs for deps, prebuilt deps should be found
     # auto-magically by GHC
+    package_dbs = []
     for cache in set.to_list(build_info.package_caches if not custom_package_caches else custom_package_caches):
         args.extend(["-package-db", cache.dirname])
+        package_dbs.append(cache.dirname)
 
-    return args
+    ghc_info = {
+        "has_version": has_version,
+        "packages": packages,
+        "package_ids": package_ids,
+        "package_dbs": package_dbs,
+    }
+    return args, ghc_info
```

Comment:
From what I can see, all info that is added to `ghc_info` here is also appended to `args` in the previous lines.

So it makes more sense to actually split this function into two incremental functions: the first collects `ghc_info` and the second appends the information to args.

This way we can assign the `ghc_info` when it is needed and don’t have to ignore `ghc_info` in most calls.

Reply:
Yeap, makes sense. Will fix.